doc/submitting-changes.xml

```diff
@@ -276,6 +276,24 @@ Additional information.
 <listitem>
 <para>When changing the bootloader installation process, extra care must be taken. Grub installations cannot be rolled back, hence changes may break people's installations forever. For any non-trivial change to the bootloader please file a PR asking for review, especially from @edolstra.</para>
 </listitem>
+
+<listitem>
+<para>If you're cherry-picking and/or commit to a stable release branch, add a
+      clear description about the reason this needs to be included in the
+      release branch.
+</para>
+<para>An example of a cherry-picked commit would look like this:</para>
+<screen>
+nixos: Refactor the world.
+
+The original commit message describing the reason why the world was torn apart.
+
+Cherry-picked-from: abcdef
+Cherry-pick-reason: I just had a gut feeling that this would also be wanted by
+                    people from the stone age.
```

Comment:
We don't need `Cherry-picked-from` because it's already provided by `git cherry-pick -x` (though I see `submitting-changes.xml` doesn't mention `-x`, so we should probably add a note that `-x` is required). So it would be something like:
```
(cherry picked from commit 54d6f1f683a448e094f31b5732c424c2467eaf8d)
Reason: bla bla bla
```
Also, providing a reason is redundant if the original message has an issue link (e.g. `Fixes #123`).